qiskit/compiler/transpiler.py

```diff
@@ -278,6 +285,23 @@ def callback_func(**kwargs):
             stacklevel=2,
         )
         return pass_manager.run(circuits, output_name=output_name, callback=callback)
+    # If a target is specified have it override any implicit selections from a backend
+    # but if an argument is explicitly passed use that instead of the target version
+    if target is not None:
+        if coupling_map is None:
+            coupling_map = target.coupling_map()
+        if basis_gates is None:
+            basis_gates = target.operation_names()
+        if instruction_durations is None:
+            instruction_durations = target.durations()
+        if inst_map is None:
+            inst_map = target.instruction_schedule_map()
+        if dt is None:
+            dt = target.dt
+        if timing_constraints is None:
+            timing_constraints = target.timing_constraints()
+        if backend_properties is None:
+            backend_properties = _target_to_backend_properties(target)
```

Comment:
This might be more natural as part of `_parse_transpile_args` (though that might mean also updating the `scheduling_method` check on L311).

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7227/commits/488eab0510f5e249b64457e608a0c7b6389f8325